### PR TITLE
drand round computed correctly

### PR DESCRIPTION
**Pull request by nikkolasg**
Fixes https://github.com/filecoin-project/lotus/issues/2170

### CODE REVIEW

File: chain/beacon/drand/drand.go [hunk 1]
```diff
@@ -190,8 +190,8 @@ func (db *DrandBeacon) VerifyEntry(curr types.BeaconEntry, prev types.BeaconEntr
 func (db *DrandBeacon) MaxBeaconRoundForEpoch(filEpoch abi.ChainEpoch, prevEntry types.BeaconEntry) uint64 {
 	// TODO: sometimes the genesis time for filecoin is zero and this goes negative
 	latestTs := ((uint64(filEpoch) * db.filRoundTime) + db.filGenTime) - db.filRoundTime
-	dround := (latestTs - db.drandGenTime) / uint64(db.interval.Seconds())
-	return dround
+	_, nextTs := dchain.NextRound(latestTs)
```
Comment (Kubuxu) [inline chain/beacon/drand/drand.go:193]: Can you make this code explicit? I am afraid of for example "bugfix" to drand causing consensus  break.

Comment (nikkolasg) [reply chain/beacon/drand/drand.go:193]: I could but the idea is that lotus should use the round - timestamp mapping that drands uses no ?

Comment (nikkolasg) [reply chain/beacon/drand/drand.go]: Done

File: chain/beacon/drand/drand.go [hunk 2]
```diff
@@ -190,8 +191,18 @@ func (db *DrandBeacon) VerifyEntry(curr types.BeaconEntry, prev types.BeaconEntr
...
+
+func timestampToRound(now int64, period time.Duration, genesis int64) uint64 {
+	if now < genesis {
+		return 1
+	}
+	fromGenesis := now - genesis
+	// we take the time from genesis divided by the periods in seconds, that
+	// gives us the number of periods since genesis. We add +1 since we want the
+	// next round. We also add +1 because round 1 starts at genesis time.
+	return uint64(math.Floor(float64(fromGenesis)/period.Seconds())) + 1
```
Comment (Kubuxu) [inline chain/beacon/drand/drand.go:205]: I see only one `+1`.

Comment (nikkolasg) [reply chain/beacon/drand/drand.go:205]: Sorry copied comment from drand - modified.

**Comment by Stebalien** [comment]
Closing because this is consensus breaking. We can pick this back up at some point in the future.

**Pull request closed without merging**